visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -50,7 +57,7 @@ public Builder(RemoteWebDriver driver, String username, String accessKey, String
       this.driver = driver;
       this.username = username;
       this.accessKey = accessKey;
-      this.endpoint = endpoint;
+      this.endpoint = resolveEndpoint(endpoint);
```

Comment:
`resolveEndpoint` should probably take a look into `driver` to see if it can infer the actual region in use.

It will prevent user setups where session runs on region `A` but Visual build is on region `B`.

Reply:
Yeah that would be ideal and simplify our constructor and usage greatly. Unfortunately there's no way to infer the actual region from `driver` at the moment.
See: https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/remote/RemoteWebDriver.html